src/main.rs

```diff
@@ -681,11 +658,48 @@ fn main() {
 mod test_program {
     use crate::{
         register::{GeneralRegister, Register, RegisterSubset},
-        Instruction, MemRegMove, Program,
+        ImmediateToRegister, Instruction, MemRegMove, Program,
     };
 
     use super::assembly::program;
 
+    #[test]
+    fn test_programs_with_different_instruction_sequences_are_not_equal() {
+        let program1 = Program {
+            bits: 64,
+            instructions: vec![],
+        };
+        let program2 = Program {
+            bits: 64,
+            instructions: vec![Instruction::ImmediateToRegister(ImmediateToRegister::Byte(
+                Register::General(GeneralRegister::D, RegisterSubset::All),
+                1,
+            ))],
+        };
+
+        assert_ne!(program1, program2);
+    }
+
+    #[test]
+    fn test_programs_with_identical_instruction_sequences_are_equal() {
```

Comment:
This is where I need to get out the property based testing Rust libraries

Reply:
This is just me trying to convince myself that the derived implementation is sensible.